Buttplug.Server.Connectors.WebsocketServer/ButtplugWebsocketServer.cs

```diff
@@ -58,28 +58,35 @@ public class ButtplugWebsocketServer
 
         public async Task StartServer([NotNull] Func<ButtplugServer> aFactory, uint maxConnections = 1, int aPort = 12345, bool aLoopBack = true, bool aSecure = false, string aHostname = "localhost")
         {
-            _cancellation = new CancellationTokenSource();
-            _serverFactory = aFactory;
+            try

```

Comment:
Why does this need a try block and an event handler? It seems like this is taking the exception throwing mechanism that async depends on, and mixing metaphors into an event handling mechanism with basically no changes otherwise. 

This is why #463 exists. Any await call that has an exception thrown in it should handle at the await level, not internally. This means that developers don't have to remember to hook up an event handler and figure out what happened where.

Reply:
The exceptions are lost when thrown from a task. Try stopping and starting the server from the GUI to see what I mean.

Reply:
AFAICT from https://github.com/buttplugio/buttplug-csharp/blob/master/Buttplug.Apps.ServerGUI/ServerControl.xaml.cs#L204, the call to StartServer in ServerControl.xaml.cs is not awaited, so yes, it is lost. 

If that was awaited (and StartServer async'd), the exception wouldn't be lost, and there is already a try/catch block at that level (which could really be reduced in scope to just the StartServer call too). That's why I filed #463, because we've got a TON of stuff around the codebase that is probably like that, and is both metaphor-mixing and non-idiomatic for async code. I stopped converting at the app level in master because my patch was already wildly out of control size-wise, so expect more weirdness there.